src/dynimage.rs

```diff
@@ -788,21 +778,20 @@ fn image_dimensions_impl(path: &Path) -> ImageResult<(u32, u32)> {
         #[cfg(feature = "hdr")]
         image::ImageFormat::HDR => hdr::HDRAdapter::new(fin)?.dimensions(),
         #[cfg(feature = "pnm")]
-        image::ImageFormat::PNM => {
-            pnm::PNMDecoder::new(fin)?.dimensions()
+        image::ImageFormat::PNM => pnm::PNMDecoder::new(fin)?.dimensions(),
+        format => {
+            return Err(image::ImageError::UnsupportedError(format!(
+                "Image format image/{:?} is not supported.",
+                format
+            )))
```

Comment:
I'd follow-up the lint by adding a `;` at the end here. Somewhat surprised that it doesn't recommend that.

Reply:
Agreed. I'm going to add `;` here.